Dockerfile

```diff
@@ -119,20 +119,41 @@ RUN git clone https://github.com/openshift/hypershift.git /hypershift
 # Build binary
 RUN OUT_DIR=/out make hypershift
 
+## Attach osdctl binary into the image
+# Setting ENV variables
+ENV OSDCTL_URL="https://github.com/openshift/osdctl/releases/latest/download/osdctl_0.46.0_Linux_x86_64.tar.gz"
+ENV OSDCTL_SHA256_CHECKSUM_URL="https://github.com/openshift/osdctl/releases/latest/download/sha256sum.txt"
+
+# Creating a working directory
+RUN mkdir -p /osdctl
+WORKDIR /osdctl
+
+# Downloading the osdctl binary
+RUN curl -sSLf -O $OSDCTL_URL
+
+# Checking the SHA-256 hash
+RUN curl -sSLf $OSDCTL_SHA256_CHECKSUM_URL | grep "osdctl_0.46.0_Linux_x86_64.tar.gz" | sha256sum -c -
```

Comment:
If you change the version you have to change this line too, factor that out a bit so you only set the version in one place.